### DISCUSSION TITLE
Issue Upgrading Strimzi from 0.21.1 to 0.31.1 in Kubernetes 1.22

### DISCUSSION THREAD

**Discussion by coylecullen** [strimzi/strimzi-kafka-operator#7503] [Q&A] [answered]
I'm running into this issue trying to upgrade strimzi to the newest version. For some background, this is what I am currently running (I go into detail in the next paragraph):
Strimzi 0.21.1
Kubernetes 1.22

After reading through documentation (https://strimzi.io/docs/operators/latest/deploying.html#con-upgrade-paths-earlier-versions-str | https://strimzi.io/docs/operators/0.24.0/deploying.html#assembly-upgrade-resources-str), I've read how in order to run the upgrades seamlessly, I need to upgrade strimzi to 0.22.0 to allow custom resources to use the v1beta2 api version, update my custom resources to use apiextensions/v1beta2, and then I would be able to upgrade strimzi to 0.31.1. However, I'm currently using kubernetes version 1.22 which isn't allowing me to upgrade to 0.22.0 due to the deprecated apiextensions/v1beta1 which is what 0.22.0 uses for it's CRDs. I think the main issue is that my kubernetes version is too new to do this upgrade process for strimzi and custom resources. I think kubernetes upgraded some strimzi resources so it would still be available to us, but they seem to be blocking any deployment of resources using deprecated apiextensions. Is there any known workaround for this? I don't see a way for us to do this upgrade without downtime so I think my only option is to upgrade directly to strimzi 0.31.1 and take the hit while I upgrade my custom resources to v1beta2/v1.

**Comment by scholzj** [comment, marked as answer]
Strimzi 0.21.1 is not compatible with Kubernetes 1.22 and if someone forced the upgrade of your Kubernetes cluster despite the operator / CRDs being incompatible, then they screwed it up. First thing to do is, you should go and and find out what the current state is. Because what you use right now si not really Strimzi 0.21 and cannot be easily upgraded. Once you know what was done to the cluster, there might be way to figure out what to do. But you should expect that you will need to test itsomewhere first - it might include things such as trying to delete everything apart from the actual Kafka pods and upgrading etc. and then installing the new Strimzi version directly etc. There was a similar discussion about this on Slack in the #strimzi channel. You should search it out and read it.

**Reply by coylecullen** [reply]
Thank you for the quick response - I had a feeling this was going to be one of those situations where things won't go smoothly haha but I believe I found the thread you were referring to in the Strimzi slack so I'll explore my options off of that. Appreciate it
